ygen/gogen.go

```diff
@@ -502,6 +502,59 @@ func (t *{{ .Receiver }}) New{{ .ListName }}(
 
 	return t.{{ .ListName }}[key], nil
 }
+`
+
+	// goListMemberRenameTemplate provides a template for a function which renames
+	// an entry within a list. It is used to generate functions for each list within
+	// a generated Go struct.
+	goListMemberRenameTemplate = `
+// Rename{{ .ListName }} renames an entry in the list {{ .ListName }} within
+// the {{ .Receiver }} struct. The entry with key oldk is renamed to newk updating
+// the key within the value.
+func (t *{{ .Receiver }}) Rename{{ .ListName }}(
+	{{- if ne .KeyStruct "" -}}
+	oldk, newk {{ .KeyStruct -}}
+  {{- else -}}
+	{{- range $key := .Keys -}}
+	oldk, newk {{ $key.Type -}}
+	{{- end -}}
+	{{- end -}}
+) error {
+	e, ok := t.{{ .ListName }}[oldk]
+	if !ok {
+		return fmt.Errorf("key %v not found in {{ .ListName }}", oldk)
+	}
+
+	ns, err := ygot.DeepCopy(e)
+	if err != nil {
+		return fmt.Errorf("cannot DeepCopy entry %v, got error: %v", oldk, err)
+	}
+	n, ok := ns.(*{{ .ListType }})
+	if !ok {
+		return fmt.Errorf("wrong type returned in list, got error: %v", err)
+	}
+	{{- if ne .KeyStruct "" -}}
+	{{- range $key := .Keys -}}
+	{{- if $key.IsScalarField }}
+	n.{{ $key.Name }} = &newk.{{ $key.Name }}
+	{{- else }}
+	n.{{ $key.Name }} = newk.{{ $key.Name }}
+	{{- end -}}
+	{{- end -}}
+	{{ else -}}
+	{{- range $key := .Keys -}}
```

Comment:
Since in this case, `len(Keys) == 1` (since we have a keyed list, and we know that there is not a composite key), changed this to `index`.